src/extension/features/budget/category-solo-mode/index.js

```diff
@@ -0,0 +1,63 @@
+import { Feature } from 'toolkit/extension/features/feature';
+import { isCurrentRouteBudgetPage } from 'toolkit/extension/utils/ynab';
+import { controllerLookup } from 'toolkit/extension/utils/ember';
+import { getToolkitStorageKey, setToolkitStorageKey } from 'toolkit/extension/utils/toolkit';
+
+export class CategorySoloMode extends Feature {
+  // ===== LISTENERS ======
+  shouldInvoke() {
+    return isCurrentRouteBudgetPage();
+  }
+
+  injectCSS() {
+    return require('./index.css');
+  }
+
+  onRouteChanged() {
+    if (this.shouldInvoke()) this.invoke();
+  }
+
+  invoke() {
+    $('.budget-toolbar').append(
+      '<span class="ember-view"><button id="all-category-expand-button">&#8597;</button><input type="checkbox" id="cat-solo-mode" class="ember-view" ' +
+        getToolkitStorageKey('catSoloMode') +
+        '><label class="ember-view button" for="cat-solo-mode">Solo Mode</label></span>'
+    );
+    $('.js-budget-table-cell-collapse').on('click', this.toggleCategory);
+    $('.budget-toolbar #cat-solo-mode').on('click', this.initializeState);
+    $('.budget-toolbar #all-category-expand-button').on('click', this.toggleAllCategories);
+    setTimeout(this.initializeState, 500);
```

Comment:
I have a concern here: 
I consider this as bad style. Unfortunately it needs to be a delay in order to get the webpage into a defined state (which it should be already regarding to the docs). If I remove the delay and 'initializeState' is directly via startup, it seems like the clickhandlers to toggle categories aren't set.
Might this be a flaw in the lifecycle management of ynab / toolkit? Is there a better way to do it?

Reply:
this is somewhat alarming if that's the case. 

I'm going to be messing around with finding a way to migrate the feature setting, i'll have a look at this at the same time. The dom should certainly be ready once we've gotten here.